pkg/binding/controller.go

```diff
@@ -509,38 +503,7 @@ func (c *Controller) reconcile(ctx context.Context, key util.Key) error {
 		return nil
 	}
 
-	if err := c.updateDecisions(obj); err != nil {
-		c.logger.Error(err, "failed to update bindingpolicy resolutions for object",
-			"object", util.RefToRuntimeObj(obj))
-		// return nil // not changing existing flow before transport is ready
-	}
-
-	//TODO (maroon): everything below this line should be deleted when transport is ready
-	clusterSet, managedByBindingPolicies, singletonStatus, err := c.matchSelectors(obj)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("error matching selectors: %s", err))
-		return nil
-	}
-
-	// if no clusters
-	if len(clusterSet) == 0 {
-		return nil
-	}
-
-	if singletonStatus {
-		clusterSet = pickSingleCluster(clusterSet)
-	}
-
-	if key.DeletedObject != nil {
-		c.logger.Info("Deleting", "object", util.RefToRuntimeObj(obj),
-			"from clusters", clusterSet)
-		deleteObjectOnManagedClusters(c.logger, c.ocmClient, *key.DeletedObject, clusterSet)
-		return nil
-	}
-
-	c.logger.Info("Delivering", "object", util.RefToRuntimeObj(obj),
-		"to clusters", clusterSet)
-	return c.deliverObjectToManagedClusters(obj, clusterSet, managedByBindingPolicies, singletonStatus)
+	return c.updateResolutions(ctx, obj)
```

Comment:
Why is it right to remove the error wrapping?

Reply:
Do you refer to the removed log line? There is logging in cases of errors in updateResolutions and its call paths, and if everything goes well, controller::processNextWorkItem has the following:
>		
            // Finally, if no error occurs we Forget this item so it does not
		// get queued again until another change happens.
		c.workqueue.Forget(obj)
		c.logger.V(2).Info("Successfully synced", "object", obj)

Reply:
No, I was referring to a change that is no longer visible. The change that I commented on replaced a return of a wrapped error with a return of an error without wrapping.

The question now is: are the errors returned by `c.updateResolutions` sufficiently self-explanator? I suspect so, given that it works on only one thing, the same object that the caller is working on.